mesonbuild/compilers/c.py

```diff
@@ -763,7 +764,7 @@ def has_function(self, funcname, prefix, env, *, extra_args=None, dependencies=N
             val = env.properties.host.get(varname, None)
             if val is not None:
                 if isinstance(val, bool):
-                    return val
+                    return val, True
```

Comment:
If it's from the cross file, it's not cached.

Reply:
I guess this is true. Changing it to `False`